src/promptflow-core/promptflow/executor/_script_executor.py

```diff
@@ -218,17 +219,34 @@ def get_inputs_definition(self):
 
     def _resolve_init_kwargs(self, c: type, init_kwargs: dict):
         """Resolve init kwargs, the connection names will be resolved to connection objects."""
+        logger.debug(f"Resolving init kwargs: {init_kwargs.keys()}.")
         sig = inspect.signature(c.__init__)
         connection_params = []
+        model_config_params = []
+        # TODO(3117908): support connection & model config from YAML signature.
         for key, param in sig.parameters.items():
             if ConnectionType.is_connection_class_name(param.annotation.__name__):
                 connection_params.append(key)
-        if not connection_params:
+            elif param.annotation.__name__ in MODEL_CONFIG_NAMES:
+                model_config_params.append(key)
+        if not connection_params and not model_config_params:
             return init_kwargs
         resolved_init_kwargs = {k: v for k, v in init_kwargs.items()}
         provider = ConnectionProvider.get_instance()
+        # parse connection
+        logger.debug(f"Resolving connection params: {connection_params}")
         for key in connection_params:
             resolved_init_kwargs[key] = provider.get(init_kwargs[key])
+        # parse model config
+        logger.debug(f"Resolving model config params: {model_config_params}")
+        for key in model_config_params:
+            model_config_val = init_kwargs[key]
+            if getattr(model_config_val, "connection", None):
+                logger.debug(f"Getting connection {model_config_val.connection} for model config.")
+                connection_obj = provider.get(model_config_val.connection)
```

Comment:
how do we handle connection not exist error?

Reply:
It'll be wrapped with User Error: 
![image](https://github.com/microsoft/promptflow/assets/7776147/fa44a58b-afea-4617-a02f-c474e6aeb123)